### PR TITLE
docs: clarify what `nvim_buf_set_name()` does

### PR DESCRIPTION
**Pull request by echasnovski**
The main motivation for this update is to provide context as to how users can watch if the buffer changed name. Currently the most reliable approach is to use `BufFilePre` and `BufFilePost` events. Their triggered after `nvim_buf_set_name()` seems to be intended.

Alternative approach: #27804

### CODE REVIEW

File: runtime/doc/api.txt [hunk 1]
```diff
@@ -2449,6 +2449,10 @@ nvim_buf_set_mark({buffer}, {name}, {line}, {col}, {opts})
 nvim_buf_set_name({buffer}, {name})                      *nvim_buf_set_name()*
     Sets the full file name for a buffer
 
+    Works similar to |:file_f|. If the buffer did have a name, that name
```
Comment (zeertzjq) [inline runtime/doc/api.txt:2452]:
```suggestion
    Works similarly to |:file_f|. If the buffer did have a name, that name
```

File: src/nvim/api/buffer.c [hunk 2]
```diff
@@ -970,6 +970,10 @@ String nvim_buf_get_name(Buffer buffer, Error *err)
 
 /// Sets the full file name for a buffer
 ///
+/// Works similar to |:file_f|. If the buffer did have a name,
```
Comment (zeertzjq) [inline src/nvim/api/buffer.c:973]:
```suggestion
/// Works similarly to |:file_f|. If the buffer did have a name,
```

File: runtime/doc/api.txt [hunk 3]
```diff
@@ -2449,6 +2449,10 @@ nvim_buf_set_mark({buffer}, {name}, {line}, {col}, {opts})
...
+    becomes the |alternate-file| name. An unlisted buffer is created to hold
+    the old name.
```
Comment (justinmk) [inline runtime/doc/api.txt:2454]: Do not repeat docs.

```suggestion
    Sets the full file name for a buffer, like |:file_f|.
```

**Review by justinmk — APPROVED** [review]

**Comment by dundargoc** [comment]
Thank you for your contribution! We collect all typo fixes and similar small fixes into a single pull request and merge it once it gets big enough: https://github.com/neovim/neovim/pull/27868

**Pull request closed without merging**